tests/click_tests/device_menu/test_pin.py

```diff
@@ -0,0 +1,213 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+from enum import Enum
+from typing import TYPE_CHECKING
+
+import pytest
+
+from ... import translations as TR
+from ..common import go_next
+from ..test_pin import (
+    PIN1,
+    PIN4,
+    _cancel_pin,
+    _delete_pin,
+    _enter_two_times,
+    _input_pin,
+    _input_see_confirm,
+    _see_pin,
+)
+from .common import (
+    Menu,
+    assert_device_screen,
+    close_device_menu,
+    enter_pin,
+    open_device_menu,
+)
+
+if TYPE_CHECKING:
+    from trezorlib.debuglink import DebugLink
+    from trezorlib.messages import Features
+
+    from ...device_handler import BackgroundDeviceHandler
+# Trezor Safe 7 only
+pytestmark = [pytest.mark.models("eckhart")]
+
+PIN_TITLE = "pin__title"
+
+
+class Situation(Enum):
+    SETUP = 1
+    CHANGE = 2
+    REMOVE = 3
+
+
+def prepare_pin_dialogue(
+    debug: "DebugLink", features: "Features", situation: Situation
+) -> None:
+    pin_title = TR.translate(PIN_TITLE)
+    assert pin_title in Menu.SECURITY.content(features)
+
+    # Open device menu
+    open_device_menu(debug)
+
+    if situation is Situation.SETUP:
+        assert features.pin_protection is False
+        # Pin submenu shouldn't be present
+        assert Menu.PIN.content(features) is None
+        # Navigate to security menu
+        Menu.SECURITY.navigate_to(debug, features)
+
+        title = TR.translate(PIN_TITLE)
+    else:
+        assert features.pin_protection is True
+        # Navigate to pin menu
+        Menu.PIN.navigate_to(debug, features)
+
+        if situation is Situation.CHANGE:
+            title = TR.pin__change
+
+        elif situation is Situation.REMOVE:
+            title = TR.pin__remove
+        else:
+            raise RuntimeError("Unknown situation")
+
+    # Trigger pin action
+    layout = debug.read_layout()
+    pin_idx = layout.vertical_menu_content().index(title)
+    debug.button_actions.navigate_to_menu_item(pin_idx)
+
+
+@pytest.mark.setup_client(uninitialized=True)
+def test_pin_uninitialized(device_handler: "BackgroundDeviceHandler"):
+    debug = device_handler.debuglink()
+    features = device_handler.features()
+
+    # Device is uninitialized, the entire security menu is not accessible
+    with pytest.raises(TypeError, match="'NoneType' is not iterable"):
+        prepare_pin_dialogue(debug, features, Situation.SETUP)
+
+
+@pytest.mark.setup_client(pin=None)
+def test_pin_setup(device_handler: "BackgroundDeviceHandler"):
+    debug = device_handler.debuglink()
+    features = device_handler.features()
+
+    prepare_pin_dialogue(debug, features, Situation.SETUP)
+
+    # Intro screen
+    go_next(debug)
+    # Enter pin twice
+    _enter_two_times(debug, PIN1, PIN1)
```

Comment:
nit: why not using `PIN4` here?
(as done in existing `test_pin_setup(device_handler`)